src/OracleService/Protocols/OracleHttpsProtocol.cs

```diff
@@ -82,7 +83,26 @@ public async Task<(OracleResponseCode, string)> ProcessAsync(Uri uri, Cancellati
                 return (OracleResponseCode.Error, message.StatusCode.ToString());
             if (!Settings.Default.AllowedContentTypes.Contains(message.Content.Headers.ContentType.MediaType))
                 return (OracleResponseCode.ContentTypeNotSupported, null);
-            return (OracleResponseCode.Success, await message.Content.ReadAsStringAsync(cancellation));
+            if (message.Content.Headers.ContentLength.HasValue && message.Content.Headers.ContentLength > OracleResponse.MaxResultSize)
+                return (OracleResponseCode.ResponseTooLarge, null);
+
+            byte[] buffer = new byte[OracleResponse.MaxResultSize];
+            var stream = message.Content.ReadAsStream(cancellation);
+            var read = await stream.ReadAsync(buffer, 0, buffer.Length, cancellation);
```

Comment:
If response stream contains more than `buffer.Length` bytes, then it's an error and we need to return `OracleResponseCode.ResponseTooLarge` code. And if I'm not mistaken, the current code will just stop reading and return a part of the response with `OracleResponseCode.Success` code (which isn't the desired behaviour).

Reply:
Please, check the way how it's done in neo-go: https://github.com/nspcc-dev/neo-go/blob/7b86a54fc0dd918a2514b0108be986a6e2cc63fa/pkg/services/oracle/response.go#L71-L82

Reply:
Please, check it again

Reply:
This part LGTM, resolved.